scarb/src/compiler/compilers/test.rs

```diff
@@ -74,6 +82,45 @@ impl Compiler for TestCompiler {
             )?;
         }
 
+        if starknet {
+            compile_contracts(test_crate_ids, target_dir, unit, db, ws)?;
```

Comment:
As in https://foundry-rs.github.io/starknet-foundry/testing/contracts-collection.html?highlight=build-external#how-contracts-are-collected, Forge requires contracts defined in dependencies to be explicitly declared in manifest file, for them to be available in tests. 


On the contrary, Cairo Test searches all dependencies (even not direct ones) for contracts and compiles all of them, without any explicit declaration. This might be sub-optimal in some cases, as contracts can be quite large (see: https://swmansion.slack.com/archives/C037B80988J/p1726501029482169). 

this only searches for contracts in the main component of the compilation unit (so the package for unit tests, and the `tests` directory for integration tests) + ones declared in `build-external-contracts`.